packages-ts/gauntlet-terra-contracts/src/commands/contracts/migration/index.ts

```diff
@@ -0,0 +1,80 @@
+import { Result } from '@chainlink/gauntlet-core'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { TerraCommand, TransactionResponse } from '@chainlink/gauntlet-terra'
+import { AccAddress, MsgMigrateContract } from '@terra-money/terra.js'
+
+type CommandInput<MsgMigrate> = {
+  contract: string
+  newCodeId: number
+  migrationMsg: MsgMigrate
+}
+
+export type MigrationMessageMaker<MsgMigrate> = (flags, args) => MsgMigrate
+
+export abstract class MigrateContract<MsgMigrate> extends TerraCommand {
+  static makeDescription = (contract) =>
+    `(NOT TESTED!) Upgrades a contract ${contract} instance to use a new Code ID. This new Code ID must expose a "migrate" function to migrate the contract state`
+  static makeExamples = (contract) => [
+    `yarn gauntlet ${contract}:migrate_contract --network=<NETWORK> --newCodeId=<CODE_ID_NUMBER> <CONTRACT_ADDRESS>`,
+    `yarn gauntlet ${contract}:migrate_contract --network=mainnet --newCodeId=2012 terra167ccv2h0z7k0p8j6qpuzwsgu5au5qvfwgmkjsl`,
```

Comment:
Do we need these specific non-general examples? I assume they are helpful to communicate types for different parameters, but what we should do instead is explicitly add typings to the framework, so help for command would list all the available flags/args and their descriptive types.

Something to consider going forward.